addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/XiaomiDeviceBaseHandler.java

```diff
@@ -0,0 +1,237 @@
+
+    private XiaomiBridgeHandler bridgeHandler;
+
+    String itemId;
+
+    Logger logger = LoggerFactory.getLogger(this.getClass().getName());
+
+    public XiaomiDeviceBaseHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        final String configItemId = (String) getConfig().get(ITEM_ID);
+        if (configItemId != null) {
+            itemId = configItemId;
+        }
+        // schedule init with random delay between 200ms and 1sec
+        scheduler.schedule(new Runnable() {
+            @Override
+            public void run() {
+                updateThingStatus();
+            }
+        }, (int) Math.max((Math.random() * 1000), 200), TimeUnit.MILLISECONDS);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Handler disposes. Unregistering listener.");
+        if (itemId != null) {
+            XiaomiBridgeHandler bridgeHandler = getXiaomiBridgeHandler();
+            if (bridgeHandler != null) {
+                bridgeHandler.unregisterItemListener(this);
+                this.bridgeHandler = null;
+            }
+            itemId = null;
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Device {} on channel {} received command {}", itemId, channelUID, command);
+        if (command.toString().toLowerCase().equals("refresh")) {
+            return;
+        }
+        execute(channelUID, command);
+    }
+
+    @Override
+    public void onItemUpdate(String sid, String command, JsonObject message) {
+        if (itemId != null && itemId.equals(sid)) {
+            logger.debug("Item got update: {}", message.toString());
+            try {
+                JsonObject data = parser.parse(message.get("data").getAsString()).getAsJsonObject();
+                parseCommand(command, data);
+            } catch (Exception e) {
+                logger.error("Unable to parse message {}", message);
+                logger.error("Caught Exeption {}", e);
+            }
+            updateThingStatus();
+        }
+    }
+
+    @Override
+    public String getItemId() {
+        return itemId;
+    }
+
+    /**
+     * @param command
+     * @param data
+     */
+    void parseCommand(String command, JsonObject data) {
+        switch (command) {
+            case "report":
+                parseReport(data);
+                break;
+            case "heartbeat":
+                parseHeartbeat(data);
+                break;
+            case "read_ack":
+                parseReadAck(data);
+                break;
+            case "write_ack":
+                parseWriteAck(data);
+                break;
+            default:
+                logger.debug("Device {} got unknown command {}", itemId, command);
+        }
+    }
+
+    /**
+     * @param data
+     */
+    void parseReport(JsonObject data) {
+        logger.debug("Got report with data: {}", data.toString());
```

Comment:
Repeated comments:
 - do not call toString in logging statement
 - please make it a single message otherwise it might be interleaved in the actual log
 - the return is not needed.